src/main/java/ee/tuleva/onboarding/mandate/email/MandateEmailSender.java

```diff
@@ -20,18 +19,11 @@
   @EventListener
   public void sendEmail(AfterMandateSignedEvent event) {
     UserPreferences contactDetails = episService.getContactDetails(event.getUser());
-    PillarSuggestion pillarSuggestion = buildPillarSuggestion(event, contactDetails);
+    ConversionResponse conversion = conversionService.getConversion(event.getUser());
+    int suggestedPillar = event.getPillar() == 2 ? 3 : 2;
```

Comment:
Maybe this logic should be encapsulated in the `PillarSuggestion`? I.e. you pass `pillar` as the constructor argument and the `suggestedPillar` is decided inside of the `PillarSuggestion` constructor.

Reply:
> Maybe this logic should be encapsulated in the `PillarSuggestion`? I.e. you pass `pillar` as the constructor argument and the `suggestedPillar` is decided inside of the `PillarSuggestion` constructor.

Also considered it, but thought it might be confusing since we don't have named params in java. But since you're in the same boat, I'll go that route, cheers.

Reply:
Or even better, just add a method called `pillarSuggestion.getSuggestedPillar()` that encapsulates that logic